baybe/surrogates/gaussian_process.py

```diff
@@ -27,7 +27,7 @@ class GaussianProcessSurrogate(Surrogate):
     # See base class.
 
     # Object variables
-    kernel: Kernel = field(factory=MaternKernel)
+    kernel: Optional[Kernel] = field(default=None)
```

Comment:
Normally I would not accept having the field optional here, but I understand that this is intertwined with the GP refactoring, so I'm fine with having it as a temporary workaround. But this will be changed back later

Reply:
Can you elaborate on that? I guess what you mean is normally you would not make this optional but use a factory which is currently not possible due to the fact that the priors are currently not chosen during initialization, or what do you mean?

Reply:
yes exactly, just answered [above](https://github.com/emdgroup/baybe/pull/219#discussion_r1579616590) 😄

Reply:
Ok, if you don't like the previous, [this](https://github.com/emdgroup/baybe/pull/219#discussion_r1579695981) will drive you crazy